### DISCUSSION TITLE
New source: Qualtrics

### DISCUSSION THREAD

**Discussion by rstuart85** [airbytehq/airbyte#35252] [New Connector Request]
## Tell us about the new integration you’d like to have
Which source and which destination? Which frequency?

Qualtrics as the source. Amazon S3 as the destination. 15 minutes frequency.

## Describe the context around this new integration
Which team in your company wants this integration, what for? This helps us understand the use case.

Customer Insights / Customer Experience team. They want the survey response data in a warehouse so they can visualise it using standard BI/EI tools.

## Describe the alternative you are considering or using
What are you considering doing if you don’t have this integration through Airbyte?

We've used their API directly for now. They also support exporting survey responses via CSV.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368090446361) by [Unito](https://www.unito.io)

**Comment by michel-tricot** [comment]
Thank you @rstuart85 for creating the issue!

are we talking about this API: https://api.qualtrics.com/api-reference/ ?

are there any models that you deem are higher priority?

**Comment by wallies** [comment]
@michel-tricot I would say the survey responses probably sits highest. I think it's done via the export api https://api.qualtrics.com/guides/docs/Guides/Common%20Tasks/getting-survey-responses-via-the-new-export-apis.md

**Comment by michel-tricot** [comment]
Thank you @wallies!

How are you doing it today?

**Comment by wallies** [comment]
@michel-tricot using the export api for surveys but as we are looking into more integration sources, we are looking into better tooling

**Comment by vitaliizazmic** [comment]
## Integration Vetting
The [integration](https://api.qualtrics.com/api-reference/docs/api-reference.md) has already been vetted

**Webhook-based? (no/partially/yes)**
No

**Available authentication modes (API key/Oauth/other)**
The Qualtrics API provides an [API token](https://api.qualtrics.com/api-reference/docs/Instructions/api-key-authentication.md) based authentication system. To authenticate using a valid token, include your token under the HTTP header X-API-TOKEN.

**Creating an account**
**Needs account.** Qualtrics REST API is a [Paid Feature](https://api.qualtrics.com/instructions/docs/Instructions/Quick%20Start/qualtrics-api-quick-start.md). The API is an additional feature. If you are unsure if your account has API access, please contact the Qualtrics Administrator at your organization. If you are the Qualtrics Administrator please contact your sales representative at Qualtrics to discuss licensing for API access.

**How to populate the account with data?**
Some streams can be populated via API, but it is needs account to answer this question.

**Available streams for sync**
Event Types
Events
Directories
  * Contact-Frequency Rules
  * Directory Contacts
  * Opted Out Directory Contacts
  * Participants
  * Sample Definitions
  * Segments
  * Transaction Batches
    ** Transactions in a Batch
 
Distributions
  * Distribution Links
 
Subscriptions

Groups
  * Users in Group

Mailing Lists
  * Bounced Contacts In Mailing List
  * Contacts In Mailing List
  * Opted-Out Contacts In Mailing List
  * Contacts
  * Samples
  
Requests
SMS Distribution
Surveys
Tickets
Users
 * User API Token

WhatsApp Distribution
 
**Integration supports incremental sync?**
Some

**Other information/blockers**
Singer Tap doesn't exist

**Comment by kyle-cheung** [comment]
+1!

**Comment by davelindley** [comment]
+1 - if credentials are needed to unblock/reproduce I can provide a dev environment to the respective devs.

**Comment by vincentkoc** [comment]
Following

**Comment by igrankova** [comment]
Hi everyone! need help here. Can't reach the support to ask for API access.

**Comment by vincentkoc** [comment]
@igrankova DM me on Airbye slack 👍, I'm an enterprise customer

**Comment by JenniWhitman** [comment]
Hey there! We have a need for this connector as well and i have a working Qualtrics account so I can take on the work to build the connector!

**Comment by igrankova** [comment]
@YowanR we have a contributor here, @JenniWhitman wants to start this connector development, the sandbox is on the way to be set up.

**Comment by YowanR** [comment]
Hi all! It's exciting to see so much enthusiasm for this connector! 
It looks like @koconder claimed this issue first so Vincent, are you planning to work on this connector? If not, then we can assign this one to @JenniWhitman.
We're doing our best on our side to get our sandbox account set up so we can test and maintain this connector going forward.

**Comment by vincentkoc** [comment]
@YowanR I would like to work on this as an Airbyte Contiburor, I have been working with Airbyte and Qualtrics partnership team to secure the API/Sandbox. I would like to finish by developing the connector if this is OK?

**Comment by JenniWhitman** [comment]
No worries @koconder ! I'll keep watching the issue so we know when it's resolved/if I can help in any way :)

**Comment by igrankova** [comment]
@koconder hi! thank you very much for help. I received an account, LastPass name: Qualtrics partner account

**Comment by igrankova** [comment]
LastPass name: Qualtrics partner account
API Token: https://airbytepartner.qualtrics.com/Q/QualtricsIdsSection/IdsSection
[API Docs](https://api.qualtrics.com/)

**Comment by YowanR** [comment]
Circling back here @koconder, are you still planning to work on this connector?

**Comment by vincentkoc** [comment]
@YowanR I have started work on this already, will try to give an ETA :)

**Comment by seanglynn-thrive** [comment]
> @YowanR I have started work on this already, will try to give an ETA :)

Hey @koconder how is the development going? I can assist if needed :)

**Comment by currents-abhishek** [comment]
hi @koconder - thanks for taking this up! Is there an ETA? Me and my team are eager to contribute if any help is needed.

**Comment by btkcodedev** [comment]
I would like to work on this issue

**Comment by btkcodedev** [comment]
Update: Waiting for test credentials from @koconder for resuming work

**Comment by vincentkoc** [comment]
@btkcodedev i cant share my own credentials, you are best to reach out to @igrankova @YowanR for test credentials if they have a partner account setup

**Comment by btkcodedev** [comment]
Sure!!, Thanks @koconder

**Comment by btkcodedev** [comment]
@igrankova Could you please send an email to Qualtrics about API token as its a paid feature. it would be useful for completing this issue

**Reply by lideke** [reply]
Hi @btkcodedev, 
Did you get the credentials to test the connector ?
If not, maybe i can run the test with our credentials

**Comment by FilahAnas** [comment]
Hello, we have worked on this connector https://github.com/airbytehq/airbyte/pull/38751 within Brevo, its currently on review.

**Reply by FilahAnas** [reply]
@lideke

**Comment by kaypro4** [comment]
Looks like this connector wasn't accepted by the Airbyte team. Apologies for the newbie question, but is there a way for me to deploy this as a low code connector to Airbyte cloud? I was able to upload [this manifest file](https://github.com/sendinblue/airbyte/blob/2468f793fb5136ef48f63d45a8dbfb0300473c52/airbyte-integrations/connectors/source-qualtrics/source_qualtrics/manifest.yaml) but I don't think that is all that needs to be done since there is no python code behind it.

**Reply by lideke** [reply]
only the manifest file is needed 
copy past into your airbyte cloud builder will be enough to have this connector
we are using OSS version

**Reply by kaypro4** [reply]
Thanks @lideke! That works great for groups, surveys, survey_questions but I get an error when testing survey_responses. 

`No module named 'source_qualtrics'`

It makes sense because that is a CustomRequestor that relies on code files located [here](https://github.com/sendinblue/airbyte/tree/2468f793fb5136ef48f63d45a8dbfb0300473c52/airbyte-integrations/connectors/source-qualtrics/source_qualtrics). My next challenge is to see if/how I can deploy those to Airbyte cloud.

**Reply by lideke** [reply]
ah yes the survey_responses stream need a parent stream so this part is in python so why it's not working with builder

**Reply by lideke** [reply]
In airbyte cloud, can you add a new docker connector ?
if yes : 
repository name : sibengineering/airbyte-source-qualtrics
docker image tag : 0.1.3

**Reply by kaypro4** [reply]
Thanks @lideke I've been trying to get an answer to that without any luck. However I did just find that "cloud" is greyed out on [this](https://docs.airbyte.com/operator-guides/using-custom-connectors) page so I believe that docker connectors are not supported on cloud. If we go OSS I'll definitely leverage your solution. Thanks again.
